ts/src/test/Exchange/test.createOrder.ts

```
@@ -0,0 +1,228 @@
+    const method = 'createOrder';
+    if (!exchange.has[method]) {
+        console.log (logPrefix, 'does not have method ', method, ' yet, skipping test...');
+        return;
+    }
+    // ensure it has cancel (any) method, otherwise we should refrain from automatic test.
+    assert (exchange.has['cancelOrder'] || exchange.has['cancelOrders'] || exchange.has['cancelAllOrders'], logPrefix + ' does not have cancelOrder|cancelOrders|canelAllOrders method, which is needed to make tests for `createOrder` method. Skipping the test...');
+
+    // pre-define some coefficients, which will be used down below
+    const limitPriceSafetyMultiplierFromMedian = 1.045; // todo: in future, if ccxt would have "maximum limit price diapason" precisions unified, we can use those coefficients, but at this moment, differet exchanges have different coefficients. for example, unlike spot-market, binance's future market has 5% boundary for limit order prices, which means you can't place limit order higher than current price * 5% (i.e. for BTC/USDT market). So, at this moment, around 5% is acceptable guaranteed price where order will get filled like market order.
+    const market = exchange.market (symbol);
+
+    // we need fetchBalance method to test out orders correctly
+    assert (exchange.has['fetchBalance'], logPrefix + ' does not have fetchBalance() method, which is needed to make tests for `createOrder` method. Skipping the test...');
+    // ensure there is enough balance of 'quote' asset, because at first we need to 'buy' the base asset
+    const balance = await exchange.fetchBalance ();
+    const initialBaseBalance = balance[market['base']]['free'];
+    const initialQuoteBalance = balance[market['quote']]['free'];
+    assert (initialQuoteBalance !== undefined, logPrefix + ' - testing account not have enough balance of' + market['quote'] + ' in fetchBalance() which is required to test ' + method);
```

I don't think we should make this check, the balance might be in a different wallet or something like that, isn't easier to let the call fail with an `InsufficientFunds` error?

---

removed